kolibri/core/assets/src/views/StorageNotification.vue

```diff
@@ -115,7 +122,11 @@
       },
       closeBanner() {
         // TODO: Store preference after closure
-        this.bannerOpened = false;
+        if (this.deviceStatus === LearnerDeviceStatus.INSUFFICIENT_STORAGE) {
+          this.bannerOpened = true;
+        } else {
+          this.bannerOpened = false;
+        }
```

Comment:
I don't quote understand this change? Could you clarify what you're intending here. It seems this would mean that if there's insufficient storage, the notification isn't dismissable

Reply:
Yes! That's was my thought.

Reply:
I did not discuss it with anyone.  I am not sure of what would be the expected behavior here. if that's the case.

Reply:
> f) It should persist the user's dismissal of the notification, using the dates of the last sync and the last removed download

So I think this connects with this deliverable above that I mentioned.

There are two primary situations where we want this notification to show:
1) The device has insufficient storage
2) User's downloads were removed

I think the text of the notification should show in that priority: if both are true, (1) should show. Either way, the notification should be dismissable. We can track the user's dismissal in the browser's local storage (see this [Vue-use composable](https://vueuse.org/core/useLocalStorage/)). So when we query the API via `useUserSyncStatus`, we should compare what's stored in local storage with those details. Specifically, we can utilize the dates of the last sync and the last removed download to know whether the user has dismissed the notification or not.

Reply:
Remember to take a look at the figma for the different use cases. I overlooked some things related to the full-facility scenarios, but I think the frontend logic should be the same regardless.
![Screenshot from 2023-06-01 08-09-41](https://github.com/learningequality/kolibri/assets/819838/a0606044-6bcf-4147-a613-8efb64d2ffdd)